### PR TITLE
feature flag to use https / container on port 4443

### PR DESCRIPTION
**Pull request by andreer**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java [hunk 1]
```diff
@@ -441,13 +447,15 @@ public URI overviewBadge(ApplicationId id) {
     /** Returns a URI of the tester endpoint retrieved from the routing generator, provided it matches an expected form. */
     Optional<URI> testerEndpoint(RunId id) {
         DeploymentId testerId = new DeploymentId(id.tester().id(), id.type().zone(controller.system()));
+        boolean useHttp = controller.system().isPublic()
+                && !directRoutingUseHttps.with(FetchVector.Dimension.APPLICATION_ID, testerId.applicationId().serializedForm()).value();
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java]:
```suggestion
                && ! directRoutingUseHttps.with(FetchVector.Dimension.APPLICATION_ID, id.tester().id().serializedForm()).value();
```

Had to try this.

**Review by jonmv — APPROVED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java [hunk 2]
```diff
@@ -441,13 +447,15 @@ public URI overviewBadge(ApplicationId id) {
     /** Returns a URI of the tester endpoint retrieved from the routing generator, provided it matches an expected form. */
     Optional<URI> testerEndpoint(RunId id) {
         DeploymentId testerId = new DeploymentId(id.tester().id(), id.type().zone(controller.system()));
+        boolean useHttp = controller.system().isPublic()
+                && ! directRoutingUseHttps.with(FetchVector.Dimension.APPLICATION_ID, id.tester().id().serializedForm()).value();
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java]:
```suggestion
         boolean useHttp =    controller.system().isPublic()
                          && ! directRoutingUseHttps.with(FetchVector.Dimension.APPLICATION_ID, id.tester().id().serializedForm()).value();
```

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java [hunk 3]
```diff
@@ -441,13 +447,16 @@ public URI overviewBadge(ApplicationId id) {
     /** Returns a URI of the tester endpoint retrieved from the routing generator, provided it matches an expected form. */
     Optional<URI> testerEndpoint(RunId id) {
         DeploymentId testerId = new DeploymentId(id.tester().id(), id.type().zone(controller.system()));
+        boolean useHttp = controller.system().isPublic()
```
Comment (andreer) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java]:
```suggestion
-        boolean useHttp = controller.system().isPublic()
```

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java [hunk 4]
```diff
@@ -441,13 +447,15 @@ public URI overviewBadge(ApplicationId id) {
     /** Returns a URI of the tester endpoint retrieved from the routing generator, provided it matches an expected form. */
     Optional<URI> testerEndpoint(RunId id) {
         DeploymentId testerId = new DeploymentId(id.tester().id(), id.type().zone(controller.system()));
+         boolean useHttp =      controller.system().isPublic()
```
Comment (andreer) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java]:
```suggestion
        boolean useHttp =      controller.system().isPublic()
```

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java [hunk 5]
```diff
@@ -441,13 +447,15 @@ public URI overviewBadge(ApplicationId id) {
...
+                           && ! directRoutingUseHttps.with(FetchVector.Dimension.APPLICATION_ID, id.tester().id().serializedForm()).value();
```
Comment (andreer) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java]:
```suggestion
                          && ! directRoutingUseHttps.with(FetchVector.Dimension.APPLICATION_ID, id.tester().id().serializedForm()).value();
```

**Review by tokle — APPROVED** [review]

**Pull request merged**